plugins/catalog-backend/src/catalog/StaticCatalog.ts

```diff
@@ -27,12 +27,19 @@ export class StaticCatalog implements Catalog {
     this._locations = locations;
   }
 
+  async addOrUpdateComponent(component: Component): Promise<Component> {
+    this._components = this._components
+      .filter((c) => c.name !== component.name)
+      .concat([component]);
```

Comment:
no need to pass in an array here, think of the compilation speedup!

Reply:
whaa but the type is supposed to be an array!

Reply:
I think it's a `...(T | T[])[]`

Reply:
but that's insane